### ISSUE TITLE
Black screen on Android ShieldTV Pro with audio.

### ISSUE THREAD

**Issue by retrobenny** [SachinVin/citra_android#77]
My screen stays completely black on my ShieldTV Pro (2015) with OTA 6.3 (Nougat,if that matters) when opening OOT3D and the game runs the title screen with audio but my whole screen is solid black with no onscreen controls or anything other than the menu elements for configuration.

Another problem is that the onscreen controls would probably make the app crash on ShieldTV just like Dolphin does but I can't know for sure,so I hope input mapping can be added so I can map my keyboard (yes,a keyboard) with it while others could map their gamepads too.

**Comment by SachinVin** [comment]
can you get me some logs they're in `internal_storage/citra-emu/logs `

**Comment by kenan1099** [comment]
Try to switch to another app and then switching back to citra
That fixes it for me on my phone (a5 2017)
I just click the "recents" button , select any app, then go back to citra.
Idk how'd you do that on Android TV, but if it doesn't have a recents tab then maybe try to go back to the home screen and going back to citra (somehow without killing the app)

**Comment by retrobenny** [comment]
Also,an image showing that it does indeed render on ShieldTV after doing the recent apps trick,so hopefully the problem can be fixed.

![oot3d recents result](https://user-images.githubusercontent.com/16418716/48049304-d23c9e00-e163-11e8-8027-01cdfaa199a6.png)

**Comment by SachinVin** [comment]
It looks like a problem similar to #72 on mali gpus, I'll see if i can find the source.
As for the controller, can you try these settings in the `internal_storage/citra-emu/config/config.ini`
but I dont have a controller to verify,

`
button_a="engine:gamepad,code:96"
button_b="engine:gamepad,code:97"
button_x="engine:gamepad,code:99"
button_y="engine:gamepad,code:100"
button_up="engine:gamepad,code:19"
button_down="engine:gamepad,code:20"
button_left="engine:gamepad,code:21"
button_right="engine:gamepad,code:22"
button_l="engine:gamepad,code:102"
button_r="engine:gamepad,code:103"
button_start="engine:gamepad,code:108"
button_select="engine:gamepad,code:109"
button_zl="engine:gamepad,code:104"
button_zr="engine:gamepad,code:105"
`

**Comment by retrobenny** [comment]
No good on control mapping. Though I am not using the Shield controller but instead a wired Logitech controller.
I am afraid to use the Shield controller now because it could break itself with an update and/or force my Shield TV to update to the buggier Oreo OTA upgrade against my will.

**Comment by SachinVin** [comment]
Are your controller buttons 1, 2, 3, 4 instead of ABXY ?
If not could you try the config without the quotes ?
like this --
`button_a=engine:gamepad,code:96`

**Comment by Priyam005** [comment]
@SachinVin please view #84

**Comment by retrobenny** [comment]
My Logitech gamepad copies standard Xbox inputs from what I see and it uses the ABXY layout.

I would focus on fixing the screen itself first then hopefully input mapping can be done in the somewhat near future to avoid the complicated manual ini configuration.
(guess I could always map my controller via Citra Windows then copy the values to test it later anyway then report back)

The game that runs for me (OOT3D) still ends up crashing into the game list UI after a short bit anyway during the menu intro,hope things can become more stable sooner rather than later.

**Comment by SachinVin** [comment]
> I would focus on fixing the screen itself first then hopefully input mapping can be done in the somewhat near future to avoid the complicated manual ini configuration.

Easy stuff first :p, I also don't have the device to debug the source of the problem so it might take a while 
> (guess I could always map my controller via Citra Windows then copy the values to test it later anyway then report back)

Likely not going to work , Windows and android use different keycodes.

> The game that runs for me (OOT3D) still ends up crashing into the game list UI after a short bit anyway during the menu intro,hope things can become more stable sooner rather than later.

Can you get me the log?

**Comment by boytoy91** [comment]
Had the same issue with the black screen switching apps fix the issue

Huawei p20 pro, android 9

**Comment by Yodakiller3000** [comment]
Yeah sometimes that happens. It just needs you to exit the game and re open it
